### PR TITLE
Start a template

### PR DESCRIPTION
**Pull request by rhpvorderman**
Most of it is @ffinfo's work. Adapted it slightly.

### CODE REVIEW

**Comment by ffinfo** [comment]
@rhpvorderman: please also add the CI files

File: library.wdl [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+        if (readgroupId != "") {
+            call readgroup.readgroup {
+                input:
+                    sampleConfigs = sampleConfigs,
+                    readgroupId = readgroupId,
+                    libraryId = libraryId,
+                    sampleId = sampleId
+            }
+        }
+    }
```
Comment (ffinfo) [inline library.wdl:52]: Some comments where to put library jobs are missing

File: pipeline.wdl [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+            jar = downloadSampleConfig.jar
+    }
+
+    scatter (sampleId in samplesConfigs.keys) {
+        call sample.sample {
+            input:
+                sampleConfigs = sampleConfigFiles,
+                sampleId = sampleId
+        }
+    }
```
Comment (ffinfo) [inline pipeline.wdl]: Some comments where to put multisample jobs are missing

File: readgroup.wdl [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+
+    call sampleConfig.SampleConfig as config {
+        input:
+            inputFiles = sampleConfigs,
+            sample = sampleId,
+            library = libraryId,
+            readgroup = readgroupId,
+            tsvOutputPath = "samples/" + sampleId + "/libs/" + libraryId + "/readgroups/" + readgroupId + "/" + readgroupId + ".config.tsv"
+    }
+
```
Comment (ffinfo) [inline readgroup.wdl]: Some comments where to put readgroup jobs are missing

File: sample.wdl [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+        if (libraryId != "") {
+            call library.library {
+                input:
+                    sampleConfigs = sampleConfigs,
+                    libraryId = libraryId,
+                    sampleId = sampleId
+            }
+        }
+    }
+
```
Comment (ffinfo) [inline sample.wdl:51]: Some comments where to put sample jobs are missing

File: sampleConfig.wdl [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+
+task SampleConfig {
```
Comment (ffinfo) [inline sampleConfig.wdl]: This file should be moved to wdl-tasks/biopet.wdl

File: sampleConfig.wdl [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+    output {
+        Array[String] keys = read_lines(stdout())
+        File? jsonOutput = jsonOutputPath
+        File? tsvOutput = tsvOutputPath
+        Object values = if (defined(tsvOutput) && size(tsvOutput) > 0) then read_map(tsvOutput) else { "": "" }
+    }
+}
+
+task DownloadSampleConfig {
+    String? release = "0.1"
```
Comment (ffinfo) [inline sampleConfig.wdl]: Maybe this download task we should not use anymore, it's not explicit enough. (sorry my fault)

File: samples.yml [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+# FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+# COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+# IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+# CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+
```
Comment (ffinfo) [inline samples.yml]: Should be rename this file to `example-samples.yml`?

**Review by ffinfo — DISMISSED** [review]

File: library.wdl (hunk 1 above)
Comment (rhpvorderman) [reply library.wdl:52]: done

File: pipeline.wdl (hunk 2 above)
Comment (rhpvorderman) [reply pipeline.wdl]: done

File: sample.wdl (hunk 4 above)
Comment (rhpvorderman) [reply sample.wdl:51]: done

File: inputs.json [hunk 8]
```diff
@@ -0,0 +1,7 @@
+{
+  "pipeline.sampleConfigFiles": ["samples.yml"],
+  "pipeline.samplesConfigs.jar_": "SampleConfig-assembly-0.1.jar",
```
Comment (DavyCats) [inline inputs.json]: I think this should be "pipeline.sampleConfigs.tool_jar" (same goes for the others)

File: inputs.json [hunk 9]
```diff
@@ -0,0 +1,7 @@
...
+  "pipeline.sample.librariesConfigs.jar_": "SampleConfig-assembly-0.1.jar",
+  "pipeline.sample.library.readgroups.jar_": "SampleConfig-assembly-0.1.jar",
```
Comment (DavyCats) [inline inputs.json]: This input doesn't exist?

**Review by DavyCats — CHANGES_REQUESTED** [review]
Looks good. I only have some comments on the input.json (see inline comments).

File: inputs.json (hunk 9 above)
Comment (rhpvorderman) [reply inputs.json]: Updated the inputs file

File: inputs.json (hunk 8 above)
Comment (rhpvorderman) [reply inputs.json]: :+1:

**Review by ffinfo — APPROVED** [review]

**Pull request merged**